pyEpiabm/pyEpiabm/sweep/queue_sweep.py

```diff
@@ -27,5 +29,20 @@ def __call__(self, time: float):
                 # Get takes person from the queue and removes them, so clears
                 # the queue for the next timestep.
                 # Update the infection status
-                person.next_infection_status = InfectionStatus.Exposed
-                person.time_of_status_change = time
+                if person.is_vaccinated:
+                    vacc_params = Parameters.instance().\
+                        intervention_params['vaccine_params']
+                    if time > (person.date_vaccinated +
+                               vacc_params['time_to_efficacy']):
```

Comment:
I think ideally this time based condition would follow a distribution such that people become protected at different times but that immunity then also wanes. Given implementing general waning immunity is probably too big a task given the timescales it might be best to leave time to efficacy as a strict delay?